pep508.nix

```diff
@@ -95,7 +95,7 @@ let
           else if stdenv.isDarwin then "darwin"
           else throw "Unsupported platform"
         );
-        platform_machine = stdenv.platform.kernelArch;
+        platform_machine = targetMachine;
```

Comment:
Does `platform_machine` support being `null`? I think a good alternative is to just throw an error instead of returning `null` for `targetMachine`

Reply:
This is probably fine, because this is only used with markers like `platform_machine == "x86_64"`, and `"null"` shouldn't match any of those. The _correct_ way to do it would be to have a complete `uname -m` mapping for every machine type supported by nixpkgs and use that value here...

Reply:
I think the best solution for now is to just add a comment indicating this and hope no one uses specific dependencies on weird architectures, because mapping every machine type in nixpkgs is definitely a bit too much.